src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -169,4 +174,91 @@ public RoleBindingList listUsersOfWorkspace() {
     return new WorkspaceManagerService(globalContext.server, currentUser)
         .getRoles(workspaceContext.getWorkspaceId());
   }
+
+  /**
+   * Lookup a controlled resource by its name. Names are unique within a workspace.
+   *
+   * @param resourceName name of resource to lookup
+   * @return the cloud resource object
+   * @throws RuntimeException if the resource is not controlled (e.g. external bucket)
+   */
+  public CloudResource getControlledResource(String resourceName) {
+    // TODO: change this method to call WSM controlled resource endpoints once they're ready
+    CloudResource cloudResource = workspaceContext.getCloudResource(resourceName);
+    if (!cloudResource.isControlled) {
+      throw new RuntimeException(resourceName + " is not a controlled resource.");
+    }
+    return cloudResource;
+  }
+
+  /**
+   * Create a new controlled resource in the workspace.
+   *
+   * @param resourceType type of resource to create
+   * @param resourceName name of resource to create
+   * @return the cloud resource that was created
+   */
+  public CloudResource createControlledResource(
+      CloudResource.Type resourceType, String resourceName) {
```

Comment:
I think we're headed towards having a different API for each resource, as they will have different parameters for creation. but also fine to start this way in this PR.
https://github.com/DataBiosphere/terra-workspace-manager/blob/ac1f4767c3ad08786ba637f7249cdb90e7131311/src/main/resources/api/service_openapi.yaml#L296

Reply:
Okay, will adapt once the WSM stuff is settled.